src/ape/managers/chain.py

```diff
@@ -389,49 +389,77 @@ class AccountHistory(BaseInterfaceModel):
     """
 
     address: AddressType
-    _transaction_list: List[ReceiptAPI] = []
+    _session_receipts: List[ReceiptAPI] = []
 
     @property
     def outgoing(self) -> Iterator[ReceiptAPI]:
         """
-        All outgoing transactions.
+        All outgoing transactions, from earliest to latest.
         """
         explorer = self.provider.network.explorer
         explorer_receipts = (
             [r for r in explorer.get_account_transactions(self.address)] if explorer else []
         )
         for receipt in explorer_receipts:
-            # TODO: Use cache
-            if receipt.txn_hash not in [t.txn_hash for t in self._transaction_list]:
+            if receipt.sender != self.address:
+                # Likely ``self.address`` is a contract.
+                # Cache the receipts by their sender instead and skip them here.
+                self.chain_manager.history.append(receipt)
+                continue
+
+            elif receipt.txn_hash not in [t.txn_hash for t in self._session_receipts]:
                 yield receipt
 
     @property
     def all(self) -> Iterator[ReceiptAPI]:
         """
-        All account transactions.
+        All account transactions, from earliest to latest.
         """
-        yield from self.outgoing
-        yield from self._transaction_list
+        sessional = sorted(self._session_receipts, key=lambda x: x.block_number)
+        for receipt in self.outgoing:
+            if not sessional or receipt.block_number <= sessional[0].block_number:
+                yield receipt
+            else:
+                yield sessional.pop(0)
 
+                # Catch up before yielding outgoing receipt.
+                if sessional:
+                    for i in range(len(sessional)):
+                        if sessional[i].block_number <= receipt.block_number:
+                            yield sessional.pop(i)
+                        else:
+                            break
```

Comment:
This seems like a good candidate for something in `ape.utils`

Reply:
Removed it for now, but we can consider that next time! We can find it again on this PR.